### ISSUE TITLE
feat: migrate styled-components

### ISSUE THREAD

**Issue by gustavopch** [nexxtway/react-rainbow#950]
I think this lib would benefit from using styled-components instead of SCSS.

-   [x] I have searched the [issues](https://github.com/nexxtway/react-rainbow/issues) of this repository and believe that this is not a duplicate.

**Is your feature request related to a problem? Please describe.**
No, not related to a problem. It's an improvement.

## Expected Behavior 🤔
Not about behavior but architecture.

## Context 🔦
By using styled-components, this lib would be able to:
- Stop having duplicated CSS and SCSS files everywhere.
- Remove className helper functions from components (`getClassNames`, `getVariantClassNames` etc) as you can inject props into styled-components.
- Stop worrying about vendor prefixes.
- Stop worrying about naming conflicts (I know BEM solves that problem, but it's even better when a problem doesn't even exist in the first place).
- Better theming support as styled-components generates CSS dynamically and theming is supported out-of-the-box with Context API.
- By the way, no need to build anything.
- Basically, a lot less lines of code which generally means less room for bugs and more productivity.

**Describe the solution you'd like**
Basically replacing CSS/SCSS by styled-components.

**Describe alternatives you've considered**
I haven't considered other alternatives.

**Comment by reiniergs** [comment]
styled-components are pretty cool, that's something we should consider, at least new components should use it, replacing CSS/SCSS by styled-components is a major refactor in our codebase at this point.

**Comment by LeandroTorresSicilia** [comment]
We will refactor all components to use styled components here the first one:
https://github.com/nexxtway/react-rainbow/pull/981

**Comment by TahimiLeonBravo** [comment]
#### Components list to migrate
- [ ] Accordion
- [ ] AccordionSection
- [ ] ActivityTimeline
- [ ] Breadcrumb
- [ ] Breadcrumbs
- [ ] Card
- [ ] Modal
- [ ] Pagination
- [ ] Sidebar
- [ ] SidebarItem
- [ ] Tab
- [ ] Tabset
- [ ] TimelineMarker
- [ ] VerticalItem
- [ ] VerticalNavigation
- [ ] VerticalSection
- [ ] VerticalSectionOverflow
- [ ] CheckboxGroup
- [ ] CheckboxToogle
- [ ] DatePicker
- [ ] GoogleAddressLookup
- [ ] Input
- [ ] Lookup
- [ ] Picklist
- [ ] PicklistOption
- [ ] RadioButtonGroup
- [ ] RadioGroup
- [ ] Rating
- [ ] Select
- [ ] Slider
- [ ] Textarea
- [ ] TimePicker
- [ ] VisualPicker
- [ ] VisualPickerOption
- [ ] VisualPickerOptionFooter
- [ ] Avatar
- [ ] AvatarMenu
- [ ] Badge (leandro)
- [ ] Button
- [ ] ButtonGroup
- [ ] ButtonIcon
- [ ] ButtonMenu
- [ ] Calendar
- [ ] Carousel
- [ ] CarouselImage
- [ ] Chip
- [ ] GMap
- [ ] MapMarker
- [ ] MenuDivider
- [ ] MenuItem
- [ ] Notification
- [ ] ProgressBar
- [ ] ProgressCircular
- [ ] ProgressIndicator
- [ ] ProgressStep
- [ ] ReCaptcha
- [ ] Spinner
- [ ] Chart
- [ ] Column
- [ ] Dataset
- [ ] Table
- [ ] TableWithBrowserPagination
- [ ] ImportRecordsFlow
